aten/src/ATen/native/TensorShape.cpp

```diff
@@ -673,11 +673,32 @@ Tensor _unsafe_view(const Tensor& self, IntList size) {
   return self.view(size);
 }
 
+static Tensor unsqueeze_sparse(Tensor const &self, int64_t dim /* should already be wrapped */) {
+  int64_t sparse_dim = self.sparse_dim();
+  auto indices = self._indices();
+  auto sizes = self.sizes().vec();
+  sizes.insert(sizes.begin() + dim, 1);
+  if (dim <= sparse_dim) {
+    auto new_indices = native::cat({
+      indices.narrow(0, 0, dim),
+      native::zeros({1, indices.size(1)}, indices.options().dtype(kLong)),
+      indices.narrow(0, dim, indices.size(0) - dim)
+    });
+    return native::sparse_coo_tensor(new_indices, self._values(), sizes, self.options());
```

Comment:
I feel like since we know `self` is already a valid sparse tensor, we can probably skip the expensive indices check at `sparse_coo_tensor`, and use ctor `_sparse_coo_tensor_with_dims_and_tensors` instead